src/Workspaces/Core/Portable/Shared/Extensions/DiagnosticDescriptorExtensions.cs

```diff
@@ -13,13 +13,22 @@ internal static class DiagnosticDescriptorExtensions
         /// 1. Compilation options from ruleset file, if any, and command line options such as /nowarn, /warnaserror, etc.
         /// 2. Analyzer config documents at the project root directory or in ancestor directories.
         /// </summary>
-        public static ReportDiagnostic GetEffectiveSeverity(this DiagnosticDescriptor descriptor, CompilationOptions compilationOptions, ImmutableDictionary<string, ReportDiagnostic> analyzerConfigSpecificDiagnosticOptions)
+        public static ReportDiagnostic GetEffectiveSeverity(this DiagnosticDescriptor descriptor,
+                                                            CompilationOptions compilationOptions,
+                                                            ImmutableDictionary<string, ReportDiagnostic> analyzerConfigSpecificDiagnosticOptions,
+                                                            ImmutableDictionary<string, ReportDiagnostic> analyzerConfigGlobalDiagnosticOptions)
         {
             var effectiveSeverity = descriptor.GetEffectiveSeverity(compilationOptions);
+            var providedOnCommandLine = compilationOptions.SpecificDiagnosticOptions.ContainsKey(descriptor.Id);
+
+            // Apply the global analyzer config severity, only if it wasn't specified via the command line
+            if (!providedOnCommandLine && analyzerConfigGlobalDiagnosticOptions.TryGetValue(descriptor.Id, out var globalReportDiagnostic))
+            {
+                effectiveSeverity = globalReportDiagnostic;
+            }
 
             // Apply analyzer config options on top of compilation options, unless the diagnostic is explicitly suppressed by compilation options (/nowarn).
```

Comment:
I think the current comment and implementation are incorrect
```suggestion
            // Apply analyzer config options on top of compilation options, they always override the compilation options and global diagnostic options.
```